gremlin/query.go

```diff
@@ -107,17 +107,19 @@ func (q QueryString) Context(list ...interface{}) QueryString {
 		if !first {
 			newQ = newQ.appends(", ")
 		}
+		first = false
 		switch t := v.(type) {
 		case time.Time:
 			if t.IsZero() {
 				return q
 			}
 			newQ = newQ.appends(fmt.Sprintf("%d", common.UnixMillis(t)))
 		case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64:
-			if t == -1 {
```

Comment:
Why calling Context with an invalid parameter that would be ignored ?

Let's not call Context with -1

Reply:
I think caller code is more elegant if I handle invalid values IsZero() or -1 within Context(), I just converted into a string and compared to "-1" so that `fmt.Sprintf("%d", t)` handles variation in types on my behalf.

Reply:
The -1 timestamp is only there to check if the query has to be run against the history or not.

I'm really not a fan of the pattern "ignore an invalid parameter".